app/controllers/api/public/v1/providers_controller.rb

```diff
@@ -13,19 +13,16 @@ def index
         end
 
         def show
-          render json: {
-            data: {
-              id: 123,
-              type: "Provider",
-              attributes: {
-                code: "ABC",
-                name: "Some provider",
-              },
-            },
-            jsonapi: {
-              version: "1.0",
-            },
-          }
+          code = params.fetch(:code, params[:provider_code])
+          provider = recruitment_cycle.providers
+                                        .find_by!(
+                                          provider_code: code.upcase,
+                                        )
+
+          render jsonapi: provider,
+                 class: { Provider: API::Public::V1::SerializableProvider },
+                 include: params[:include],
+                 fields: { providers: provider_fields }
```

Comment:
not something made in this PR but `provider_fields` should return nil if nothing is specified. The default behaviour seems to serialize everything defined in the serializer if it is nil, therefore there is no need to have a default provider fields passed in.

Reply:
Done, thanks.